src/screens/external-content.tsx

```
@@ -0,0 +1,228 @@
+  ConnectedDispatchProps &
+  StackScreenProps<
+    {
+      ExternalContent: {
+        contentRef: string;
+        contentType: ExternalTypeState;
+      };
+      Modals: {screen: 'LevelEnd'; params: {isCorrect: boolean; progressionId: string}};
+    },
+    'ExternalContent'
+  >;
+
+const External: React.FC<Props> = ({
+  route,
+  navigation,
+  externalContent,
+  brand,
+  updateState: updateState_,
+  getOneLoginTokenForContent: getOneLoginTokenForContent_,
+  completeProgressionAndPersist: completeProgressionAndPersist_,
+  fetchAdditionalInfo: fetchAdditionalInfo_,
+  getRemoteProgressionAndPersist: getRemoteProgressionAndPersist_,
+}) => {
+  const {contentRef, contentType} = route.params;
+  const expectedContentUrl = `${brand?.host}/externalContent/${contentRef}`;
+  const isEmptyUrl = externalContent.contentUrl === '';
+  const animationValue = React.useRef(
+    new Animated.Value(externalContent.validateButtonStatus === 'hidden' ? 0 : 104),
+  ).current;
+
+  const finishCourse = React.useCallback(() => {
+    try {
+      completeProgressionAndPersist_(externalContent.progressionId, () => {
+        const levelEndParams = {
+          isCorrect: true,
+          progressionId: externalContent.progressionId,
+        };
+        navigation.navigate('Modals', {screen: 'LevelEnd', params: levelEndParams});
+      });
+    } catch (error) {
+      navigation.goBack();
+    }
+  }, [completeProgressionAndPersist_, navigation, externalContent.progressionId]);
+
+  function onDidFinishCourseButtonPress() {
+    updateState_({contentStatus: 'finished'});
+  }
+
+  function onNavStateChange(navState: WebViewNavigation) {
+    if (navState.url.startsWith(expectedContentUrl) && navState.url.endsWith('/end')) {
+      updateState_({contentStatus: 'finished'});
+    }
+    if (navState.url.startsWith(expectedContentUrl)) {
+      try {
+        getRemoteProgressionAndPersist_(contentRef);
+      } catch (error) {
+        // won't do nothing for the moment
```

we might want to do something in these empty catch clauses